infra/terraform/modules/data_access/github_webhooks_handler/tests/conftest.py

```diff
@@ -0,0 +1,140 @@
+from hashlib import sha1
+from moto import mock_sns
+import boto3
+import hmac
+import json
+import os
+
+import pytest
+
+
+#################
+#  Tests state  #
+#################
+
+
+@pytest.fixture
+def state():
+    return {
+        "event": {
+            "headers": {
+                "X-Hub-Signature": "signature",
+                "X-GitHub-Event": "team",
+            },
+            "body": "{\"a JSON string\": true}",
+        }
+    }
+
+
+###########
+#  Event  #
+###########
+
+
+@pytest.fixture
+def event(state):
+    return state["event"]
+
+
+@pytest.fixture
+def event_type(event):
+    return event["headers"]["X-GitHub-Event"]
+
+
+###############
+#  Signature  #
+###############
+
+
+@pytest.fixture
+def given_invalid_signature(state):
+    state["event"]["headers"]["X-Hub-Signature"] = "sha3=invalid"
+
+
+@pytest.fixture
+def given_valid_signature(state, secret):
+    message = "hello this is dog"
+    signature = sha1_signature(message, secret)
```

Comment:
There's no need to compute this signature every run - if the message remains constant, you can precompute the signature and hardcode it here. It's a minor optimization from a performance point of view, but it removes a bunch of code, which is always a good thing.